src/main/java/io/github/opencubicchunks/cubicchunks/core/asm/mixin/ICubicWorldSettings.java

```diff
@@ -44,4 +44,8 @@
     boolean isCubic();
 
     void setCubic(boolean cubic);
+    
+    ResourceLocation getCompatibilityGeneratorType();
+    
+    void setCompatibilityGeneratorType(ResourceLocation compatibilityGeneratorType);
```

Comment:
This does not need to go in there ad in level.dat. Store it in per-dimension WorldSavedData, and when there is no WorldSavedData, take it from config. The reason it's not that way for isCubic is that we would have to know when it is a new world being created.